vosekast_control/Scale.py

```diff
@@ -119,45 +123,28 @@ def start_measurement_thread(self):
 
     # diagnostics
     def print_diagnostics(self):
-        if self.emulate:
-            self.logger.info("Printing diagnostics:")
-            self.logger.info("self.threads: " + str(self.threads))
-            self.logger.info("Thread loop alive: " +
-                str(self.thread_loop.is_alive()))
-            self.logger.info("Thread readscale alive: " +
-                            str(self.thread_readscale.is_alive()))
-            self.logger.info("self.is_running = " + str(self.is_running))
-            self.logger.info("constant_tank_ready: " +
-                            str(self.vosekast.constant_tank.is_filled))
-            self.logger.info("measuring_tank_ready: " + str(self.vosekast.measuring_drain_valve.is_closed
-                                                            and not self.vosekast.measuring_tank.is_filled))
-            self.logger.info("constant_pump_running: " +
-                            str(self.vosekast.pump_constant_tank.is_running))
-            self.logger.info("measuring_drain_valve.is_closed: " +
-                            str(self.vosekast.measuring_drain_valve.is_closed))
-            self.logger.info("measuring_tank.is_filled: " +
-                            str(self.vosekast.measuring_tank.is_filled))
-        else:
-            self.logger.info("Printing diagnostics:")
-            self.logger.info("self.threads: " + str(self.threads))
-            self.logger.info("self.connection.is_open: " +
-                            str(self.connection.is_open))
-            self.logger.info("Thread loop alive: " +
-                            str(self.thread_loop.is_alive()))
-            self.logger.info("Thread readscale alive: " +
-                            str(self.thread_readscale.is_alive()))
-            self.logger.info("self.is_running = " + str(self.is_running))
-            self.logger.info("constant_tank_ready: " +
-                            str(self.vosekast.constant_tank.is_filled))
-            self.logger.info("measuring_tank_ready: " + str(self.vosekast.measuring_drain_valve.is_closed
-                                                            and not self.vosekast.measuring_tank.is_filled))
-            self.logger.info("constant_pump_running: " +
-                            str(self.vosekast.pump_constant_tank.is_running))
-            self.logger.info("measuring_drain_valve.is_closed: " +
-                            str(self.vosekast.measuring_drain_valve.is_closed))
-            self.logger.info("measuring_tank.is_filled: " +
-                            str(self.vosekast.measuring_tank.is_filled))
-
+        self.logger.info("Diagnostics:" +
+        "\nself.threads: " + str(self.threads) +
```

Comment:
usually the `\n` is on the end of the line and not on the beginning of the new one.

Reply:
but then I would have to format it like this:
 "self.threads: " + str(self.threads)  + "\n" +

Reply:
Correct. Alternative you can use format strings:
`f"self.threads: {str(self.threads)}\n"`
Try to bring so all logs in one multiline log.